standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5026,7 +5026,8 @@ static boolean isMustPurge(EnvironmentContext envContext, Table tbl) {
     //  3. Either
     //    3.1. User has specified PURGE from the commandline, and if not,
     //    3.2. User has set the table to auto-purge.
-    return ((envContext != null) && Boolean.parseBoolean(envContext.getProperties().get("ifPurge")))
+    return (envContext != null && envContext.getProperties() != null
```

Comment:
why is this change?
 public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }